component/block_cache/block_cache.go

```diff
@@ -932,6 +935,16 @@ func (bc *BlockCache) WriteFile(options internal.WriteFileOptions) (int, error)
 
 	// log.Debug("BlockCache::WriteFile : Writing handle %v=>%v: offset %v, %v bytes", options.Handle.ID, options.Handle.Path, options.Offset, len(options.Data))
 
+	// check for possible random write scenario and block the write request
+	blockIndex := bc.getBlockIndex(uint64(options.Offset))
+	if int64(blockIndex*bc.blockSize) < options.Handle.Size-int64(MIN_WRITE_BLOCK*bc.blockSize) {
```

Comment:
Lets say we have n blocks, this condition will be true for first n-3 blocks, does that mean we don't allow write on those blocks even if they are sequential?

Reply:
Updated the logic